src/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs

```diff
@@ -97,6 +99,32 @@ internal static class KnownHeaders
         public static readonly KnownHeader XRequestID = new KnownHeader("X-Request-ID");
         public static readonly KnownHeader XUACompatible = new KnownHeader("X-UA-Compatible");
 
+        public static readonly IReadOnlyList<KnownHeader> TrailerDisallowedHeaders = Array.AsReadOnly(new[]    // rfc7230 4.1.2.
```

Comment:
Can this be made a `HashSet` so we don't need the extra copy here and in `HttpConnection`?

Reply:
TrailerDisallowedHeader is a public static field whose contents must not be changed, so it's better to have a readonly collection here to avoid accidental modifications. Unfortunately, there is no IReadOnlySet interface, so I decided to use IReadOnlyList.

Reply:
It's public but on an internal class.  And there are plenty of other such things on this type where our own internal consuming code is expected to do the "right thing", e.g. if you look at KnownHeader, you'll see several exposed arrays where consumers shouldn't change the contents.  We should prioritize perf here.

Reply:
Ok, I got the point, but anyways having a HashSet<KnownHeader> is not enough for my new logic in HttpResponseHeaders because known headers might get converted to custom ones which are compared only by names. Thus, I cannot simply re-use HashSet<KnownHeader>, but have to create a HashSet<string> with a custom comparer.

Reply:
This list is only being used to initialize these two `HashSet`:

https://github.com/dotnet/corefx/blob/db05f70e41c4187e1ad44c700f078f45c1e9e3cb/src/System.Net.Http/src/System/Net/Http/Headers/HttpResponseHeaders.cs#L21

https://github.com/dotnet/corefx/blob/db05f70e41c4187e1ad44c700f078f45c1e9e3cb/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L45

The ask here is that we either:
a) get rid of both hashsets and use a flag as suggested in https://github.com/dotnet/corefx/pull/42229/files#r340850684
b) if we want to keep both hashsets, use the `HashSet<KnownHeader>` to initialize the `HashSet<string>`.

And either way, we should get rid of this list.

Reply:
>We should prioritize perf here

I agree with @stephentoub on this point. Since this is internal code, we should add more Assert's if we think it will help us call the code correctly.

Reply:
Thanks.